lib/conduit/test.ex

```diff
@@ -76,28 +76,16 @@ defmodule Conduit.Test do
 
       MyApp.Broker.publish(:message, %Conduit.Message{})
       assert_message_published(:message)
+      message_name = :message
+      assert_message_published(message_name)
 
   """
   defmacro assert_message_published(name) when is_atom(name) do
     quote do: assert_received({:publish, _, unquote(name), _, _, _})
   end
 
-  defmacro assert_message_published(message_pattern) do
-    message_code = Macro.to_string(message_pattern)
-
-    quote do
-      assert_received({:publish, _, var!(name), unquote(message_pattern), _, _})
-
-      IO.warn("""
-      Calling assert_message_published/1 with a message pattern is deprecated. Replace with:
-
-          assert_message_published #{inspect(binding()[:name])}, #{unquote(message_code)}
-
-      Or
-
-          assert_message_published #{inspect(binding()[:name])}
-      """)
-    end
+  defmacro assert_message_published(name) do
+    quote do: assert_received({:publish, _, ^unquote(name), _, _, _})
```

Comment:
The `^` should be up to the user. So, they could do something like:

``` elixir
assert_message_received name
assert name == :foo

#or

name = :foo
assert_message_received ^name
```

Reply:
good point, I've tried this yesterday but I messed up something...
done!